csrc/scheduler/normalization_utils.h

```diff
@@ -212,5 +212,34 @@ getOptionalInnerOuterPersistentBufferBatches(
 using ReductionType = reduction_scheduler_utils::ReductionType;
 ScheduleHeuristic getPersistentHeuristicFor(ReductionType reduction_type);
 
+// Check the operations and input tensors of the fusion. This
```

Comment:
Why are these three functions organized in this way? Looks somewhat random to me. For example, when I first looked at `checkViewBufferTopology`, I thought it had something to do with the buffer for reshape tensors because of the function name.

About half of the `checkReductionPattern` function is only for the combined scheduler. Does it make sense to unify them?

Reply:
> Why are these three functions organized in this way? Looks somewhat random to me. For example, when I first looked at `checkViewBufferTopology`, I thought it had something to do with the buffer for reshape tensors because of the function name.

The original `canScheduleCompileTime` function has been broken down into three steps, **based on the original order of checks where simpler checks come first**.
(1) Initial checks related to operations and inputs that are common across all persistent schedulers have been grouped into the `checkOpsAndInputs` function.
(2) Subsequently, a vector of reduction TensorViews is extracted and used to verify reduction types and patterns, including specialized checks for the innerOuter scheduler.
(3) Additional validations related to view operations, persistent buffers, and fusion topology have been consolidated into a new function named `checkViewOpsPersistentBufferFusionTopology`.
 
If we break down these three functions, it would essentially revert to the original `canScheduleCompileTime` function without altering the checking orders.

> About half of the `checkReductionPattern` function is only for the combined scheduler. Does it make sense to unify them?

The 2nd part of that function is moved to `InnerOuterPersistentKernelScheduler::canScheduleCompileTime`

Reply:
Not quite sure what you mean, but it's just unnatural to me to have something like `checkViewOpsPersistentBufferFusionTopology`. The three checks inside the function are independent. Why not having three smaller functions? Why do you think they should be combined into the single function?

Reply:
I attempted to consolidate these 3 functions into a single function to eliminate redundancy.
After combing into a single function, the `InnerOuterPersistentKernelScheduler::canScheduleCompileTime` can directly use it instead of calling 3 checks separately. 

But as you said, there are no connections among these 3 checks and they are independent. So, maybe leaving them as standalone functions is more reasonable. `checkViewOpsPersistentBufferFusionTopology` is removed.